R/summary.pop_data.R

```diff
@@ -28,57 +32,65 @@ summary.pop_data <- function(object, strata = NULL, ...) {
     object %>%
     distinct(across(all_of(cols)))
 
-  cat("\nn =", nrow(ages), "\n")
-
-  cat("\nDistribution of age: \n\n")
-
   age_summary <-
     ages %>%
     summarise(
+      .by = all_of(strata),
+      n = n(),
       min = min(.data[[age_column]]),
       first_quartile = quantile(.data[[age_column]], 0.25),
       median = median(.data[[age_column]]),
       mean = mean(.data[[age_column]]),
       third_quartile = quantile(.data[[age_column]], 0.75),
-      max = max(.data[[age_column]]),
-      .by = all_of(strata)
-    ) %>%
-    print()
-
-  cat("\nDistributions of antigen-isotype measurements:\n\n")
+      max = max(.data[[age_column]])
+    )
 
   ab_summary <-
     object %>%
     dplyr::summarize(
       .by = all_of(c("antigen_iso", strata)),
-      Min = object %>%
-        get_value() %>%
-        min(na.rm = TRUE),
-      `1st Qu.` = object %>%
-        get_value() %>%
-        quantile(.25, na.rm = TRUE),
-      Median = object %>%
-        get_value() %>%
-        median(),
-      `3rd Qu.` = object %>%
-        get_value() %>%
-        quantile(.75, na.rm = TRUE),
-      Max = object %>%
-        get_value() %>%
-        max(na.rm = TRUE),
-      `# NAs` = object %>%
-        get_value() %>%
-        is.na() %>%
-        sum()
-    ) %>%
-    as.data.frame() %>%
-    print()
-
-  to_return <- list(
-    n = nrow(ages),
-    age_summary = age_summary,
-    ab_summary = ab_summary
-  )
-
-  return(invisible(to_return))
+      across(
+        .cols = all_of(value_column),
+        .fns = list(
+          Min = ~ min(.x, na.rm = TRUE),
+          `1st Qu.` = ~ quantile(.x, p = .25, na.rm = TRUE),
+          Median = ~ median(.x, na.rm = TRUE),
+          `3rd Qu.` = ~ quantile(.x, p = .75, na.rm = TRUE),
+          Max = ~ max(.x, na.rm = TRUE),
+          `# NAs` = ~ is.na(.x) %>% sum()
+        ),
+        .names = "{.fn}"
```

Comment:
I rewrote this section to fix an issue where the data wasn't being stratified correctly; see below, in the "before" output, all of the rows of `ab_summary` were identical:

# before:

<img width="806" alt="image" src="https://github.com/UCD-SERG/serocalculator/assets/2474437/dddc9f97-b680-4f7a-aa81-9c26ac66c7bd">

# after:

<img width="827" alt="image" src="https://github.com/UCD-SERG/serocalculator/assets/2474437/515ea22e-fcfb-44b2-b0a6-56ae9a0cecbf">